### PR TITLE
Implement stackdriver exporter

### PR DESCRIPTION
**Pull request by dazuma**
The exporter converts spans to protos and issues API calls in a background thread pool.

### CODE REVIEW

**Comment by dazuma** [comment]
Actually hold off on this for a moment. The code as it stands assumes a task ordering for the client creation: in a single-thread pool, the client creation task must get scheduled before the first export task, or the pool will deadlock. This _should_ hold, but better to be safe and use promises to ensure the correct scheduling. I'll commit a fix shortly.

**Comment by dazuma** [comment]
Fix pushed.

File: test/exporter_test.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    # responds appropriately.
+    def mock_client.nil?; false; end
+
+    converter = OpenCensus::Trace::Exporters::Stackdriver::Converter.new project_id
+    expected_span_protos = [span1, span2].map { |span| converter.convert_span span }
+    mock_client.expect :batch_write_spans, nil, ["projects/my-project", expected_span_protos]
+
+    exporter = OpenCensus::Trace::Exporters::Stackdriver.new \
+      project_id: project_id,
+      mock_client: mock_client
```
Comment (chingor13) [inline test/exporter_test.rb:51]: Test is failing because you need to mock the credentials also or make the credentials loading lazy/promise.

**Review by chingor13 — APPROVED** [review]
Other than the test failure, the code looks great.

**Review by chingor13 — APPROVED** [review]

**Comment by dazuma** [comment]
\o/ tests are finally passing.

**Review by chingor13 — APPROVED** [review]

**Pull request merged**